.github/workflows/go_lint.yml

```diff
@@ -0,0 +1,28 @@
+name: Go lint
+
+on:
+  push:
```

Comment:
We don't use the `push` entry here, but rather protect the repository against direct pushes to `master` (which should be renamed to `main` :) and only allow merges when the tests pass.

This avoids running an unnecessary lint when you merge.